### PR TITLE
Replace lodash.get with local function; support array paths.

### PR DESCRIPTION
**Pull request by davidtheclark**
@olsonpm or @sudo-suhas for review, please.

cc @stof, @cspotcode

### CODE REVIEW

**Comment by cspotcode** [comment]
Looks good to me. 👍

**Comment by olsonpm** [comment]
I've got things goin on til sunday - will review then.

File: src/getPropertyByPath.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// @flow
+'use strict';
+
+// Resolves property paths defined with period-delimited strings or arrays of strings.
+// Property names that include periods are only understand in array paths.
```
Comment (olsonpm) [inline src/getPropertyByPath.js]: Had a moment to browse now - this differs from the `_.get` implementation which special cases a string to first test for the key before splitting on a period.

For backward comaptibility (even before _.get was introduced) we could do the same e.g.
```js
// getPropertyByPath.js

// Resolves property paths defined with period-delimited strings or arrays of strings.
function getPropertyByPath(source: Object, path: string | Array<string>): any {
  // first attempt the path as though it were a single key regardless whether it has periods
  if (typeof path === 'string' && source.hasOwnProperty(path))
    return source[path];

  const parsedPath = typeof path === 'string' ? path.split('.') : path;
  let config = source;
  for (const key of parsedPath) {
    if (!config || !config.hasOwnProperty(key))
      return;

    config = config[key];
  }

  return config;
}
```

and

```js
// getPropertyByPath.test.js

const source = {
  'ant.man': 'superhero',
}

test('attempts the string key before splitting', () => {
  expect(getPropertyPath(source, 'ant.man')).toBe(source['ant.man']);
});
```

Comment (davidtheclark) [reply src/getPropertyByPath.js]: Good point, @olsonpm. I added a commit to address this.

File: README.md [hunk 2]
```diff
@@ -420,6 +422,18 @@ Use periods to describe a path to nested properties. For example, the value `'co
 }
 ```
 
+If property names within the path include periods, you need to use an array of strings. For example, the value `['configs', 'foo.bar', 'baz']` will get you the `"baz"` value in a \`package.json like this:
```
Comment (olsonpm) [inline README.md]: this is such an edge case that I doubt it matters, but with the additional comment maybe change this to

```
If **nested** property names within the path include periods
```

Comment (davidtheclark) [reply README.md]: Yes, this is a very ugly edge case. But sure I'll make that change.

Comment (olsonpm) [reply README.md]: after that I think we're good then.  If people get confused for some reason we can amend wording later.

Thanks for coding this.

**Pull request merged**